### PR TITLE
fix wheel name

### PR DESCRIPTION
**Pull request by zpcore**
Previous PR merged too fast, this is the one I tested with the output shown in https://github.com/pytorch/xla/pull/7882#issuecomment-2297734061.

### CODE REVIEW

**Review by JackCaoG — COMMENTED** [review]
what's the `\3/` for?

**Comment by zpcore** [comment]
> what's the `\3/` for?

This is the replacement for group 3. (`item.prefix` value is `torch_xla` in our case. )

The original text `torch_xla-2.5.0+git0e35022+nightly-cp310-cp310-linux_x86_64.whl` is split into three groups:
1) from beginning to the `+` sign;
2) any characters starting with `+` until `-cp`;
3) `-cp` until the end of the string. This is captured as group 3 (\3).

It basically reuses `-cp310-cp310-linux_x86_64.whl` in the final output.

File: infra/ansible/roles/build_srcs/tasks/main.yaml [hunk 1]
```diff
@@ -76,12 +76,12 @@
...
     cp {{ item.dir }}/*.whl .
     # First rename to add '.dev'
-    rename -v "s/^{{ item.prefix }}-(.*?)(\+.*?)-cp/{{ item.prefix }}-\1.dev-cp/" *.whl
+    rename -v 's/^{{ item.prefix }}-(.*?)(\+.*?)(-cp.*$)/{{ item.prefix }}-\1.dev\3/' *.whl
     mv /tmp/staging-wheels/* /dist/
     popd
 
     # Second rename to append the date in YYYYMMDD format
-    rename -v "s/^{{ item.prefix }}-(.*?)-cp/{{ item.prefix }}-\1.dev$(date -u +%Y%m%d)-cp/" *.whl
+    rename -v "s/^{{ item.prefix }}-(.*?)(\+.*?)(-cp.*$)/{{ item.prefix }}-\1.dev$(date -u +%Y%m%d)\3/" *.whl
```
Comment (JackCaoG) [inline infra/ansible/roles/build_srcs/tasks/main.yaml:84]: I don't fully understand why `-cp` is replaced with `\3/`. what's the `(\+.*?)` trying to match?

Can we just add a comment in the yaml file to indicate the origional whl name and what it suppose to update it to?

Comment (zpcore) [reply infra/ansible/roles/build_srcs/tasks/main.yaml:84]: This one probably tells better about the grouping: https://regex101.com/r/5Uldrj/1. Let me add details in the file.

**Review by JackCaoG — APPROVED** [review]
I will approve to unblock but please add some comments as suggested and merge it after that.

**Comment by zpcore** [comment]
Here is the naming change summary:

We want to make two name updates: 
- `torch-2.5.0-cp310-cp310-linux_x86_64.whl` -->  
`torch-2.5.0.dev-cp310-cp310-linux_x86_64.whl` and `torch-2.5.0.devYYYYMMDD-cp310-cp310-linux_x86_64.whl`;
- `torch_xla-2.5.0+gitac7fd44-cp310-cp310-linux_x86_64.whl` -->  
`torch_xla-2.5.0.dev-cp310-cp310-linux_x86_64.whl` and `torch_xla-2.5.0devYYYYMMDD-cp310-cp310-linux_x86_64.whl`.

To handle both cases, we can use regex:  
First rename: `s/^(.*?)\-(.*?)(\+([^ -]+))?\-(.+)/\1-\2.dev-\5/`  
Second rename: `s/^(.*?)\-(.*?)(\+([^ -]+))?\-(.+)/\1-\2.dev$(date -u +%Y%m%d)-\5/`  

Basically the filename is split into 5 groups to concatenate:  
group 1: prefix stop at first -: `torch_xla` or `torch`;    
group 2: anything after first - before next + or -: `2.5.0`;  
group 3 (can be empty): any thing after + before next -: `+gitac7fd44`;  
group 4 (can be empty): group 3 without +: `gitac7fd441`;   
group 5: anything from next - to the end: `cp310-cp310-linux_x86_64.whl`.  
This should make name into `{1}-{2}.dev-{5}` and `{1}-{2}.dev{date}-{5}`.

**Comment by JackCaoG** [comment]
thanks @zpcore do you want to merge this pr so you have anything pending?

**Comment by zpcore** [comment]
Naming looks fine after testing locally with pip==24.2. I will check tomorrow's build and update the README accordingly if everything looks good.
